File: ClimbEdge.Application/DTOs/OrganizationDTO.cs
using ClimbEdge.Domain.Enums.Organizations;

namespace ClimbEdge.Application.DTOs
{
    public record class CreateOrganizationDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? DisplayName { get; set; }
        public string? Description { get; set; }
        public OrganizationType Type { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? Website { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string Country { get; set; } = string.Empty;
        public string? PostalCode { get; set; }
        public string? TimeZone { get; set; }
        public string? LogoUrl { get; set; }
        public DateTime? FoundedDate { get; set; }
    }

    public record class UpdateOrganizationDTO
    {
        public string? DisplayName { get; set; }
        public string? Description { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? Website { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? PostalCode { get; set; }
        public string? TimeZone { get; set; }
        public string? LogoUrl { get; set; }
        public string? BannerUrl { get; set; }
        public bool? IsPublic { get; set; }
    }

    public record class GetOrganizationDTO : BaseDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? DisplayName { get; set; }
        public string? Description { get; set; }
        public OrganizationType Type { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? Website { get; set; }
        public string? City { get; set; }
        public string Country { get; set; } = string.Empty;
        public bool IsVerified { get; set; }
        public bool IsPublic { get; set; }
        public string? LogoUrl { get; set; }
        public string? BannerUrl { get; set; }
    }

    public record class AddOrganizationMemberDTO
    {
        public long OrganizationId { get; set; }
        public long UserId { get; set; }
        public MembershipType MembershipType { get; set; } = MembershipType.Basic;
    }

    public record class RemoveOrganizationMemberDTO
    {
        public long OrganizationId { get; set; }
        public long UserId { get; set; }
    }

    public record class GetOrganizationMemberDTO : BaseDTO
    {
        public long OrganizationId { get; set; }
        public long UserId { get; set; }
        public MembershipType MembershipType { get; set; }
        public MembershipStatus Status { get; set; }
        public string? Role { get; set; }
        public DateTime JoinedAt { get; set; }
        public DateTime? ExpiresAt { get; set; }
    }

    public record class CreateOrganizationEventDTO
    {
        public long OrganizationId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string EventType { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? Location { get; set; }
        public long? FacilityId { get; set; }
        public int? MaxParticipants { get; set; }
        public DateTime? RegistrationDeadline { get; set; }
        public decimal? Cost { get; set; }
        public string? Currency { get; set; }
        public bool RequiresRegistration { get; set; } = false;
        public bool IsPublic { get; set; } = true;
        public string? SkillLevelRequired { get; set; }
        public string? Instructor { get; set; }
        public string? ContactEmail { get; set; }
        public string? ContactPhone { get; set; }
    }

    public record class UpdateOrganizationEventDTO
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Location { get; set; }
        public int? MaxParticipants { get; set; }
        public DateTime? RegistrationDeadline { get; set; }
        public string? Status { get; set; }
        public bool? IsPublic { get; set; }
    }

    public record class GetOrganizationEventDTO : BaseDTO
    {
        public long OrganizationId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string EventType { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? Location { get; set; }
        public int? MaxParticipants { get; set; }
        public int CurrentParticipants { get; set; }
        public decimal? Cost { get; set; }
        public string? Currency { get; set; }
        public bool IsPublic { get; set; }
        public string Status { get; set; } = string.Empty;
    }

    public record class RegisterEventParticipantDTO
    {
        public long EventId { get; set; }
        public long UserId { get; set; }
        public string? Notes { get; set; }
    }

    public record class GetOrganizationEventParticipantDTO : BaseDTO
    {
        public long EventId { get; set; }
        public long UserId { get; set; }
        public DateTime RegistrationDate { get; set; }
        public string Status { get; set; } = string.Empty;
        public string? PaymentStatus { get; set; }
        public DateTime? CheckInTime { get; set; }
    }
}

File: ClimbEdge.App/shared/src/types/OrganizationDTO.ts
import { BaseDTO } from './BaseDTO';

export enum OrganizationType {
  Club = 0,
  Company = 1,
  Federation = 2,
  School = 3,
  Other = 4,
}

export enum MembershipType {
  Basic = 0,
  Standard = 1,
  Premium = 2,
  Honorary = 3,
}

export enum MembershipStatus {
  Active = 0,
  Inactive = 1,
  Suspended = 2,
  Expired = 3,
}

// Organization
export interface CreateOrganizationDTO {
  name: string;
  displayName?: string;
  description?: string;
  type: OrganizationType;
  email?: string;
  phone?: string;
  website?: string;
  address?: string;
  city?: string;
  state?: string;
  country: string;
  postalCode?: string;
  timeZone?: string;
  logoUrl?: string | null;
  foundedDate?: string | null;
}

export interface UpdateOrganizationDTO {
  displayName?: string;
  description?: string;
  email?: string;
  phone?: string;
  website?: string;
  address?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  timeZone?: string;
  logoUrl?: string | null;
  bannerUrl?: string | null;
  isPublic?: boolean;
}

export class GetOrganizationDTO extends BaseDTO {
  name!: string;
  displayName?: string;
  description?: string;
  type!: OrganizationType;
  email?: string;
  phone?: string;
  website?: string;
  city?: string;
  country!: string;
  isVerified!: boolean;
  isPublic!: boolean;
  logoUrl?: string | null;
  bannerUrl?: string | null;
}

// Members
export interface AddOrganizationMemberDTO {
  organizationId: number;
  userId: number;
  membershipType: MembershipType;
}

export interface RemoveOrganizationMemberDTO {
  organizationId: number;
  userId: number;
}

export class GetOrganizationMemberDTO extends BaseDTO {
  organizationId!: number;
  userId!: number;
  membershipType!: MembershipType;
  status!: MembershipStatus;
  role?: string;
  joinedAt!: string;
  expiresAt?: string | null;
}

// Events
export interface CreateOrganizationEventDTO {
  organizationId: number;
  title: string;
  description?: string;
  eventType: string;
  startDate: string;
  endDate: string;
  location?: string;
  facilityId?: number | null;
  maxParticipants?: number;
  registrationDeadline?: string | null;
  cost?: number;
  currency?: string;
  requiresRegistration: boolean;
  isPublic: boolean;
  skillLevelRequired?: string;
  instructor?: string;
  contactEmail?: string;
  contactPhone?: string;
}

export interface UpdateOrganizationEventDTO {
  title?: string;
  description?: string;
  startDate?: string;
  endDate?: string;
  location?: string;
  maxParticipants?: number;
  registrationDeadline?: string | null;
  status?: string;
  isPublic?: boolean;
}

export class GetOrganizationEventDTO extends BaseDTO {
  organizationId!: number;
  title!: string;
  description?: string;
  eventType!: string;
  startDate!: string;
  endDate!: string;
  location?: string;
  maxParticipants?: number;
  currentParticipants!: number;
  cost?: number;
  currency?: string;
  isPublic!: boolean;
  status!: string;
}

export interface RegisterEventParticipantDTO {
  eventId: number;
  userId: number;
  notes?: string;
}

export class GetOrganizationEventParticipantDTO extends BaseDTO {
  eventId!: number;
  userId!: number;
  registrationDate!: string;
  status!: string;
  paymentStatus?: string;
  checkInTime?: string | null;
}

export interface OrganizationFilters {
  publicOnly?: boolean;
  country?: string;
  page?: number;
  pageSize?: number;
}
